mobile/ios/ios-core.ts

```diff
@@ -980,30 +983,80 @@ export class PlistService implements Mobile.IiOSDeviceSocket {
 	}
 }
 
-export class GDBServer implements Mobile.IGDBServer {
-	private socket: Mobile.IiOSDeviceSocket  = null;
+util.inherits(GDBStandardOutputAdapter, stream.Transform);
+function GDBStandardOutputAdapter(opts?: stream.TransformOptions) {
+	stream.Transform.call(this, opts);
+	this.utf8StringDecoder = new string_decoder.StringDecoder("utf8");
+}
+GDBStandardOutputAdapter.prototype._transform = function(packet: Buffer, encoding: string, done: Function): void {
+	var result = "";
 
-	constructor(private service: number,
-		private $injector: IInjector) {
+	for (var i = 0; i < packet.length; i++) {
+		if (packet[i] === "$".charCodeAt(0)) {
+			var start = ++i;
 
-		if(hostInfo.isWindows()) {
-			this.socket = this.$injector.resolve(WinSocket, {service: this.service, format: 0});
-		} else if(hostInfo.isDarwin()) {
-			this.socket = this.$injector.resolve(PosixSocket, {service: this.service, format: CoreTypes.kCFPropertyListXMLFormat_v1_0});
+			while (packet[i] !== "#".charCodeAt(0)) {
+				i++;
+			}
+			var end = i;
+
+			// Skip checksum
+			i++;
+			i++;
+
+			if (!(packet[start] === "O".charCodeAt(0) && packet[start + 1] !== "K".charCodeAt(0))) {
+				continue;
+			}
+			start++;
+
+			var hexString = packet.toString("ascii", start, end);
+			var hex = new Buffer(hexString, "hex");
+			result += this.utf8StringDecoder.write(hex);
+		}
+	}
+
+	done(null, result);
+};
+
+util.inherits(GDBSignalWatcher, stream.Writable);
+function GDBSignalWatcher(opts?: stream.WritableOptions) {
+	stream.Writable.call(this, opts);
+}
+GDBSignalWatcher.prototype._write = function(packet: Buffer, encoding: string, callback: Function) {
+	for (var i = 0; i < packet.length - 2; i++) {
+		if (packet[i] === "$".charCodeAt(0) && (packet[i + 1] === "T".charCodeAt(0) || packet[i + 1] === "S".charCodeAt(0))) {
+			// SIGKILL
+			if (packet[i + 2] === "9".charCodeAt(0)) {
+				process.exit(1);
+			}
 		}
 	}
+	callback(null);
+};
+
+export class GDBServer implements Mobile.IGDBServer {
+	constructor(private socket:net.Socket) {
+	}
 
 	public run(argv: string[]): void {
 		this.send("QStartNoAckMode");
-		this.socket.sendMessage("+");
+		this.socket.write("+");
 		this.send("QEnvironmentHexEncoded:");
 		this.send("QSetDisableASLR:1");
 
-		var encodedArguments = _.map(argv, (arg, index) => util.format("%d,%d,%s", arg.length*2, index, new Buffer(arg).toString("hex"))).join(",");
-		this.send("A"+encodedArguments);
+		var encodedArguments = _.map(argv, (arg, index) => util.format("%d,%d,%s", arg.length * 2, index, new Buffer(arg).toString("hex"))).join(",");
+		this.send("A" + encodedArguments);
 
 		this.send("qLaunchSuccess");
-		this.send("vCont;c");
+
+		if(options.logging) {
+			this.socket.pipe(new (<any>GDBStandardOutputAdapter)()).pipe(process.stdout);
```

Comment:
I checked the type definitions, and it seems that pipe is declared ```pipe<T extends NodeJS.WritableStream>(destination: T, options?: { end?: boolean; }): T;``` I would try with \<NodeJS.WritableStream\>